### ISSUE TITLE
Crouching Under Surfaces

### ISSUE THREAD

**Issue by xEvoGx** [ElectricNightOwl/ArmSwinger#37]
( possibly linked to #31 )

I seem to come up with these questions when I'm away from my headset, in Starbucks on my laptop lol...

...does/will your system support being able to crouch under surfaces (hiding under desks, crawling through vents), and if you stand up in said enclosed areas will you be pushed back to the last safe location?

Thanks kjack!

**Comment by kjack9** [comment]
No.  Currently, ArmSwinger doesn't handle clipping into surfaces above you well at all.

Furthermore, it's very difficult to determine what positions are "safe" while the user is crawling.  Do we do it based on the starting headset height?  Have a height measurement feature?  Something else?

One thing I could do is make is make it so that if you stand up into a surface, rather than rewinding you ArmSwinger simply fades out the view until you stop clipping the surface.  Would that work well for you?

**Comment by xEvoGx** [comment]
Yea **[VRTK](https://github.com/thestonefox/SteamVR_Unity_Toolkit)** using _last standing position_ for their system, including measuring height. I suggest checking out **017_CameraRig_TouchpadWalking** to see how they implement it. Long story short, they generate a box collision volume starting at the headset, and going down to the ground (the PlayArea).

As for your last comment, fading is already implemented in VRTK, and works _well enough_ but I've become a fan of keeping the player from clipping at all if possible.

**Comment by xEvoGx** [comment]
Oh and before you ask _"why don't you just use VRTK's systems"_, I'm always on the lookout for different implementations. :smile:

**Comment by kjack9** [comment]
I'll take a look at how VRTK is handling this.  My first inclination is to implement a fading system to keep from further complicating ArmSwinger, but will do my research first.

**Comment by xEvoGx** [comment]
Well sure you may want one regardless, yes. It's funny how we all got behind teleporting which has helped mitigate a lot if not all of these issues, and now we're faced with how to actually create _RealMovement.tm_ all over again from scratch within these unpredictable systems (people moving about lol).

**Comment by kjack9** [comment]
Duplicate of #31 as well.
